### ISSUE TITLE
[FEATURE REQUEST] Option for "Just go to folders for series"

### ISSUE THREAD

**Issue by zumbawumba** [JerloPH/HomeCinema#12]
Have an option to just index the series, without indexing the episodes or seasons. As such, HomeCinema would still show the poster for the series, but clicking it will just open the folder for where the video files are stored.

Sample directory structure:
D:\TV Series\Title of Series\ > (all the files for that series)

This option would allow users to name and sort the video files for their series however they want. This also solves the issues with trying to match all those video files with the various online databases, which all too often have mistakes and/or are incomplete, and don't always match each other. I'm not the first to suggest such a thing, as users of Plex, Kodi, and Jellyfin have all complained about this.

Now, HomeCinema could handle all series this way, but having it as an option is likely best. I don't care either way, and leave that for you to consider.

That said, for years I've used this method in EMDB, where when I click a poster for a series, and it simply opens the folder for that series, which has worked great for me. Yet understand that EMDB does not have that feature/option, but I found a workaround to achieve it.

**Comment by JerloPH** [comment]
As of writing, the current release doesn't index TV series per episode. The way it works is that, you provide a top-level directory that contains folders named with series title. Inside those series title folders are the video files. This way, the app only recognizes the folder name and index it, searching TMDB for information. 

On the second note, you can right-click an Entry and click 'Open in Explorer'. This will load Windows Explorer, with the folder of the series highlighted and selected.

**Comment by zumbawumba** [comment]
My mistake, as I actually have not tested series in HC yet and assumed it worked otherwise, even after reading all about HC.

That said, please don't remove this feature in future releases. Also, when I say "series" that means both so-called TV series and Anime series, and I'm assuming you mean the same.

**Comment by zumbawumba** [comment]
I reopened this, as what I requested here does not work for me, as you said it should:

"..This way, the app only recognizes the folder name and index it, searching TMDB for information."

Thus, perhaps this is a bug, but I don't think that's the case. Regardless, I pointed HomeCinema v0.7.1 to my TV folder with series in it, each with their own folder, and each with at least one video in them, as HC won't index anything without a video in them (FYI).

Sample directory structure:
F:\TV\Proper Title of Series + year(s)\ > video files

This is the result:

![BUG  TV FAIL](https://user-images.githubusercontent.com/86828379/125028208-7a554400-e03c-11eb-8842-779aac2d7521.jpg)

So in this case, HC did indeed index these episodes (video files) and did not display the folder names, and most received no data such as posters and such, and those that did mistakenly got detected as movies.

Thus my original request here still remains, but I'll make it more clear here:

Have an option to just index the series folders, without indexing the episodes/files or seasons. As such, HomeCinema would show the poster for the series, and left-clicking it will just open the folder for where the video files are stored.

In other words, I just want a poster for each series, with quick access to the folders of where they are. With my series, I don't want HC to index thousands of videos.

That all said, if what I'm experiencing is a bug, or I'm overlooking something, then please advise.

Now, if this is an issue with HC not being able to detect my TV folders because they are not named "proper", then that would lead me to request a feature that would detect them as they are, as I'm not naming those folders in some unconventional way. And being that none were detected proper, that consideration seems rather moot at this point, but there you go.

**Comment by JerloPH** [comment]
Read "Adding Media Files" section of the README file.

Put "F'\TV" in the 'Series Location'.

NOTE: You need to remove the prev entries so it won't show, the one on the screenshots

**Comment by zumbawumba** [comment]
Got it.. testing again. I guess I'm an idiot.  😢

**Comment by zumbawumba** [comment]
In the settings if I just select for series, save and restart, I get nothing:

![no go](https://user-images.githubusercontent.com/86828379/125054441-0e360880-e05b-11eb-8a2a-e7f1daea3e99.jpg)

If I select for "series" and "media files" locations to that same folder I then get TV posters, but it still loads all those mkv files too, which is not what I wanted:

![wtf1](https://user-images.githubusercontent.com/86828379/125061222-f44bf400-e061-11eb-82ad-09c4df00bb62.jpg)
![wtf2](https://user-images.githubusercontent.com/86828379/125061240-f9a93e80-e061-11eb-8322-a2476b9f2d5c.jpg)

I then removed the "media files" location but I still get the same as the image above here.

Also, if I just use that wizard when HC starts: Browse For Folder > Select folder to search for media files, then that just gives me that result farther above in that pic with all mkv files and no TV posters. Thus I've been canceling that step for the most part.

I've tried everything I can think of, other than trying it on a different PC.

Besides all that, with the TV posters I do get, they don't open that folder, as it opens them all to my main TV series folder. Also, one of them opens to My PC directory for some reason. Hmm. Regardless, if you could get HC to open the specific folder, then that would be another step in what I'm looking for.

So in part, this kind of works as advertised, but I'm still getting all these issues.

What does work, I like what I see, and I appreciate all the hard work you have done. I know HC is still in beta, thus I know the gig. Thanks again! 😃

P.S. I don't use HC with the actual TV/movie files, as I actually write reviews and make YouTube videos, and thus I use HC to simply index my work. 😉

**Comment by JerloPH** [comment]
First, try a fresh copy and use that. 

And make sure the directory structure follow this scheme:

```F:\TV Series\Title of it\Episode 01.mp4```

And place the path : ``F:\TV Series`` on **Series Locations**.

As a side note, anything on **Media Locations** will search all subfolders and only locate video files whose extensions are in the **Media File Format**. That's why putting the path there lists all episodes. **Series Locations** however, **only indexes the folders inside the root path provided**.

For the 2 bugs regarding opening the folder:

1. If it opens the main path (your root directory), it means the index path exists, but cannot select it for some reasons (might be locked or lacking privilege to that folder).
2.  If it opens **My PC**, this mean that the path your trying to open is not existing. 

Also, if you can replicate the same issue, compress and zip the folder **'logs'** located on the app directory, and send them to me. Do not re-open HomeCinema after replicating the issue since it might delete those logs (You can however turn this feature off in the Settings before trying to replicate).

Here's how to manually change the setting:

1. Go to **[App directory]\data** and open **settings.json** with Notepad or any Text Editor.
2. Find this line: ``"autoClean": 1,``, and change the value to **0**.

**Comment by zumbawumba** [comment]
> First, try a fresh copy and use that.

That's what I had been doing, and then I tried clearing it and restating as you had suggested before, but no success.

> And make sure the directory structure follow this scheme..

Yup, did that too. I didn't do that all at first, but after that I made sure the directory structure was just as that with my TV folder named "TV Series".

> ..Media Locations will search all subfolders and only locate video files whose extensions are in the Media File Format. That's why putting the path there lists all episodes. Series Locations however, only indexes the folders inside the root path provided.

Thanks for making that clear, as that confirms what I had always expected, but as I displayed above in those pics, I only get folders added to HC when P:\TV Series is added under "Series Location" **AND** P:\TV Series is added again under "Media Locations".  In other words, after trying most everything, this was a fluky work-around I found that finally got my folders detected, yet that result was not ideal.

> ..bugs regarding opening the folder:
> might be locked or lacking privilege to that folder

I was thinking that might be the case. In fact, that might be the reason for all my problems here, but IDK. I'm going to test HC on another Win10 PC, which doesn't have the latest MS updates, and is not running any optional security or drive monitoring software. If all works on that other PC, I will then give it the latest MS update, and start adding my other software to it. Thus, if some other software is causing issue, then I hope to track it down and report back to you. 

> If it opens My PC, this mean that the path your trying to open is not existing.

Sure. FYI: with that fluke entry, when I right-click the poster and go to File Paths, it properly displays the folder path:
"P:\TV Series\Tim & Eric Awesome Show, Great Job! (2007-2017) ★"
I just checked again, and the folder is indeed there on P still.
I will keep that complete HC test in another folder with that fluke, in case we need to go back to it. I will now see if I can repeat this glitch again in another test. For now, my only theory is that something in HC doesn't like that folder name. 

BTW, I see and understand that HC doesn't do well with fetching data from the web when the folder name is as this:
A Daft TV Show (2007-2017)
As this is the proper format it likes:
A Daft TV Show (2007)
But that's another issue, of little concern to me right now, just so you know.

> Also, if you can replicate the same issue, compress and zip the folder 'logs' located on the app directory, and send them to me. Do not re-open HomeCinema after replicating the issue since it might delete those logs (You can however turn this feature off in the Settings before trying to replicate).

Yup. We are on the same page here. I will do that as well.

**Comment by JerloPH** [comment]
> Sure. FYI: with that fluke entry, when I right-click the poster and go to File Paths, it properly displays the folder path:
"P:\TV Series\Tim & Eric Awesome Show, Great Job! (2007-2017) ★"
I just checked again, and the folder is indeed there on P still.
I will keep that complete HC test in another folder with that fluke, in case we need to go back to it. I will now see if I can repeat this glitch again in another test. For now, my only theory is that something in HC doesn't like that folder name.
 
Yep, the issue might be character-related. I don't know if I'll work on it tho since most folder names doesn't use special characters. For now, this is a confirmed bug

> BTW, I see and understand that HC doesn't do well with fetching data from the web when the folder name is as this:
A Daft TV Show (2007-2017)
As this is the proper format it likes:
A Daft TV Show (2007)
But that's another issue, of little concern to me right now, just so you know.

This is intentional on my part. Basically, I want the app to parse the first occurrence of length-4 digits as a year. Cause that's the most common way of naming conventions but I'll try to add an special case for ####-#### format.

**Comment by zumbawumba** [comment]
I replicated that issue, but don't forget, that issue is happening under my workaround method involved with that other issue, as detailed here above. Thus exploring this fluke seems rather moot to me. Regardless, here you are:

So without going into great detail, I started again fresh and still replicated that issue, with this:

P:\TV Series\Tim and Eric Awesome Show, Great Job! (2017)\Tim & Eric 1x01.mkv

Notice that this time I changed Tim **&** Eric to Tim **and** Eric, but the issue still appeared, where left-clicking the poster opens up My PC. 

I then tried again, but this time I added this: 
P:\TV Series\Tim and Eric Awesome Show Great Job (2017)

That fixed it. For some reason, HC did not like that name with those punctuations before.

That all said, my main issue remains, where HC does not detect my series folders at all, unless I use that funky workaround of mine.

**Comment by zumbawumba** [comment]
> This is intentional on my part. Basically, I want the app to parse the first occurrence of length-4 digits as a year. Cause that's the most common way of naming conventions but I'll try to add an special case for ####-#### format.

I understand, and that's common among all the software like HC I have tested. It Just seems silly to me to have just the first year in the folder name, for a series that may have been on for 20 or more years, ha.

My main problem is, is that HC won't fetch the data for it if it's named like this:
A Daft TV Show (2007-2017)

Also, I have many series that are still on, which I keep in a folder named TV active, which are named as this:
More of That Daft TV Show (2018-)

I guess for the data fetch search, for the year, if HC could detect (#### and ignore everything after that, then perhaps that might help. 

Just be prepared for names like this:
She Came From the Year 3001 (1984)
As such names can spring issues, at least with my experience with regular expressions, which isn't much. I learned enough to clean up most all my TV and movie file names, with the help of Advanced Renamer.

Also, some devs of such software don't even use the year for their search fetches. In fact, with movies in HC, that appears to be the case, as I had several movies with the same name but different years, but they all got detected as the same movie. But that's yet another issue I'm not concerned about right now, as I'm still testing other things, ha.

**Comment by JerloPH** [comment]
The year parsing is only relevant when the IMDB fetching isn't successful (entry might not exist or wrong spelling), as the year will be replaced by whatever IMDB provides.  The mechanics behind this is that only the title is searched. If by any chance, any year is included on the searching feature, I will look into it and fix it.

**Comment by zumbawumba** [comment]
> The year parsing is only relevant when the IMDB fetching isn't successful.

Then that makes since, as with what I likewise described, say you have 3 movies of the same name but of different years, then if IMDB finds/matches a name for each of them, it will use that and ignore the years of those 3 movies, correct? Thus in my case, it gave me the same movie for those 3 different movies. 

I'm used to such errors, as I know there are limits to what can be practically addressed with code.

> If by any chance, any year is included on the searching feature, I will look into it and fix it.

Sounds great. Just keep in mind that there is perhaps no simple fix. For example, if you always have it search/fetch for the title+year, then that introduces other problems, as you might imagine. As such, one solution is to have it search for just the given title, and if it finds more than one result, it then asks the user to pick from a list of those results. Beyond that, if it finds more than one result for a title, it could then look to match the year, and at least put that on the top of the list for the user to choose from. Or, after the process is done, just have a log that the user can go over, which would flag such exceptions. Beyond that, I guess some devs have tried to list all the fluky exceptions in this process, but that seems like a burden best left to AI.

That all said, this post has gone a bit off topic, ha. Thus, don't forget:

As it is, my [FEATURE REQUEST] Option for "Just go to folders for series" here is still a thing, which has been further explained, somewhere up there, ha.

As for the errors I've experienced, I will open separate bug reports as need be, after my tests are done. Agreed?

BTW, I almost forgot, did you still want those logs I promised?

**Comment by JerloPH** [comment]
Yep, send those logs here, or email me.

**Comment by JerloPH** [comment]
Folder of series will now open on Explorer, instead of just highlighting it. 
Will close now
